### PR TITLE
Add route, service, and functionality for Admin Invitations

### PR DESCRIPTION
**Pull request by ji-mmyliu**
## Notion ticket link

[Invite Admin](https://www.notion.so/uwblueprintexecs/Invite-Admin-8b898b398018424dab08f27cc6f60adc)

## Implementation description
* Route POST endpoint added to authRoutes.ts
* Checks for request authorization and request integrity through middleware
* Creates new user with temporary password generated by 'generate-password' npm library
* Sends email to prompt for user account verification

## Steps to test
1. Use Postman to attempt to post to `/auth/inviteAdmin`. Get the bearer token from the frontend (details in Notion ticket).
2. Should receive email and be able to log in as new username and the temporary password given in the email.

## What should reviewers focus on?
* Making sure unauthorized and bad requests are caught as errors by the backend

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/rest/authRoutes.ts
Comment (98ZhaoJeffrey) [inline backend/rest/authRoutes.ts:1]: how did u guys install generate-password but it didn't update the package.json?

Comment (98ZhaoJeffrey) [inline backend/rest/authRoutes.ts:1]: run the linter, everything else is good

**Review by 98ZhaoJeffrey — APPROVED** [review]
add the password generate to package.json, run linter and it's good

**Review by carolynzhang18 — COMMENTED** [review]
Nailed it!! Tested a couple different things (invalid body params, bad access token, logging in with wrong password) and everything works as expected 🧑‍🍳

My only suggestion would be to push the email-sending logic back into the service layer, as a method in `authService.ts`. We could put the stuff from line 141 to 156 into a new method `authService.sendAdminInvite`, and then call that method from the `/inviteAdmin` function. This helps us declutter the routes file, and you'll also notice all the other email-sending logic in the backend (e.g. `sendEmailVerificationLink`) is in the service layer as well. What do you guys think?

**Comment by ji-mmyliu** [comment]
> Nailed it!! Tested a couple different things (invalid body params, bad access token, logging in with wrong password) and everything works as expected 🧑‍🍳
> 
> My only suggestion would be to push the email-sending logic back into the service layer, as a method in `authService.ts`. We could put the stuff from line 141 to 156 into a new method `authService.sendAdminInvite`, and then call that method from the `/inviteAdmin` function. This helps us declutter the routes file, and you'll also notice all the other email-sending logic in the backend (e.g. `sendEmailVerificationLink`) is in the service layer as well. What do you guys think?

Ooh yes that's a good point! We're on it, chief :saluting_face:

**Comment by 98ZhaoJeffrey** [comment]
lgtm

File: backend/services/implementations/authService.ts [hunk 1]
```diff
@@ -164,6 +164,38 @@ class AuthService implements IAuthService {
     }
   }
 
+  async sendAdminInvite(
+    email: string,
+    temporaryPassword: string,
+  ): Promise<void> {
+    if (!this.emailService) {
+      const errorMessage =
+        "Attempted to call sendEmailVerificationLink but this instance of AuthService does not have an EmailService instance";
```
Comment (carolynzhang18) [inline backend/services/implementations/authService.ts:173]: `sendAdminInvite`? :))

**Review by carolynzhang18 — APPROVED** [review]
image

File: backend/services/implementations/authService.ts (hunk 1 above)
Comment (ji-mmyliu) [reply backend/services/implementations/authService.ts:173]: oopsie daisy :upside_down_face:

**Pull request merged**